src/h3lib/lib/algos.c

```diff
@@ -813,6 +817,58 @@ H3Error _getEdgeHexagons(const GeoLoop *geoloop, int64_t numHexagons, int res,
     return E_SUCCESS;
 }
 
+/**
+ * Returns true if the given cell is in the polygon using the containment
+ * mode flags.
+ */
+bool cellInPolygon(H3Index hex, const GeoPolygon *geoPolygon,
+                   const BBox *bboxes, uint32_t flags) {
+    if ((flags & FLAG_ANY_VERTEX) == FLAG_ANY_VERTEX) {
+        CellBoundary hexBoundary;
+        H3_EXPORT(cellToBoundary)(hex, &hexBoundary);
+
+        for (int i = 0; i < hexBoundary.numVerts; i++) {
+            if (pointInsidePolygon(geoPolygon, bboxes, &hexBoundary.verts[i])) {
+                return true;
+            }
+        }
+
+        LatLng hexCenter;
+        H3_EXPORT(cellToLatLng)(hex, &hexCenter);
+
+        if (pointInsidePolygon(geoPolygon, bboxes, &hexCenter)) {
```

Comment:
As below - this is faster than the boundary check, move to the beginning of the block?